projects/vdk-plugins/vdk-meta-jobs/src/vdk/plugin/meta_jobs/cached_data_job_executor.py

```diff
@@ -127,3 +133,78 @@ def get_all_jobs(self):
 
     def get_currently_running_jobs(self):
         return [j for j in self._jobs_cache.values() if j.status in ACTIVE_JOB_STATUSES]
+
+    def start_new_job_execution(self, job_name: str, team_name: str) -> str:
+        """
+        Start a new data job execution.
+        The stages of the process are:
+            1) Get the latest available execution_id before any new
+               executions have been started.
+            2) Start a new execution.
+            3) In case of a Timeout exception, check if a new execution
+               has been started and re-try step 2 if needed.
+
+        NOTE: A loop is used to handle situations, where due to some
+        network instability, a socket timeout happens. The execution of
+        the data job may have started, but we don't know because an execution
+        id was not returned due to the exception.
+
+        :param job_name: name of the data job to be executed
+        :param team_name: name of the owning team
+        :return: id of the started job execution
+        """
+        current_retries = 0
+        execution_id = None
+
+        latest_available_execution_id = self.get_latest_available_execution_id(
+            job_name=job_name, team=team_name
+        )
+
+        while current_retries < ALLOWED_RETRIES:
+            try:
+                execution_id = self._executor.start_job(job_name, team_name)
+                return execution_id
+            except TimeoutError as e:
+                log.info(
+                    f"A timeout exception occurred while starting the {job_name} data job. "
+                    f"Exception was: {e}"
+                )
+
+            if not execution_id:
```

Comment:
I don't see how this is not always true? 
Because if it is set then it is retuned above on line 166?